### PR TITLE
Add negative tests for unsafe encoding and decoding

### PR DESCRIPTION
**Pull request by hughbe**
/cc @tarekgh

### CODE REVIEW

File: src/System.Text.Encoding/tests/System.Text.Encoding.Tests.csproj [hunk 1]
```diff
@@ -12,8 +12,10 @@
   </PropertyGroup>
   <!-- Default configurations to help VS understand the configurations -->
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```
Comment (tarekgh) [inline src/System.Text.Encoding/tests/System.Text.Encoding.Tests.csproj]: why you need to repeat adding this property for every configuration? you can just add it globally for the whole project so we don't have to duplicate it for every configuration

File: src/System.Text.Encoding/tests/NegativeEncodingTests.cs [hunk 2]
```diff
@@ -36,11 +36,19 @@ public static void GetByteCount_Invalid(Encoding encoding)
             Assert.Throws<ArgumentOutOfRangeException>("chars", () => encoding.GetByteCount(new char[3], 2, 2));
             Assert.Throws<ArgumentOutOfRangeException>("chars", () => encoding.GetByteCount(new char[3], 3, 1));
             Assert.Throws<ArgumentOutOfRangeException>("chars", () => encoding.GetByteCount(new char[3], 4, 0));
+
+            char[] chars = new char[3];
+            fixed (char* pChars = chars)
+            {
+                char* pCharsLocal = pChars;
```
Comment (tarekgh) [inline src/System.Text.Encoding/tests/NegativeEncodingTests.cs]: do we really need to declare this? can't we just use pChars?

File: src/System.Text.Encoding/tests/NegativeEncodingTests.cs [hunk 3]
```diff
@@ -85,11 +93,34 @@ public static void GetBytes_Invalid(Encoding encoding)
...
+            char[] chars = new char[3];
+            byte[] bytes = new byte[3];
+            byte[] smallBytes = new byte[1];
+            fixed (char* pChars = chars)
+            fixed (byte* pBytes = bytes)
+            fixed (byte* pSmallBytes = smallBytes)
+            {
+                char* pCharsLocal = pChars;
+                byte* pBytesLocal = pBytes;
+                byte* pSmallBytesLocal = pSmallBytes;
```
Comment (tarekgh) [inline src/System.Text.Encoding/tests/NegativeEncodingTests.cs]: same as previous comment

File: src/System.Text.Encoding/tests/NegativeEncodingTests.cs (hunk 2 above)
Comment (hughbe) [inline src/System.Text.Encoding/tests/NegativeEncodingTests.cs]: I get a compiler error if I use `pChars` as I can't use something in a fixed statement in a lambda expression (i.e. the `Assert.Throws<...>(() => ...)`)

Comment (tarekgh) [inline src/System.Text.Encoding/tests/NegativeEncodingTests.cs]: right, I forgot we cannot use the fixed variables inside the lambda expressions or anonymous methods.

**Comment by tarekgh** [comment]
other than my comment on the csproj file, LGTM. thanks @hughbe

File: src/System.Text.Encoding/tests/System.Text.Encoding.Tests.csproj (hunk 1 above)
Comment (hughbe) [inline src/System.Text.Encoding/tests/System.Text.Encoding.Tests.csproj]: done

**Pull request merged**